src/shogun/classifier/AveragedPerceptron.cpp

```diff
@@ -89,7 +89,12 @@ void CAveragedPerceptron::iteration()
 		}
 		linalg::update_mean(w, cached_w, num_prev_weights);
 		linalg::update_mean(bias, cached_bias, num_prev_weights);
+
+		observe<SGVector<float64_t>>(get_step(), "w");
```

Comment:
Regarding the AveragedPerceptron, I have used directly the `observe` method since we are using a linalg call to update both w and the bias. I thought it did not make any sense to use `put()` here since the values are updated in place (and it is more efficient). I think that in many other places around Shogun we will find these kinds of situations.  What do you think? @karlnapf

Reply:
Yep I think that is good. As long as the algorithm is updating its members rather than some temporary variable.

Reply:
I am confused about the call, wouldnt you have to pass the `w` vector to the `observe` call? Or does this call mean "observe a model parameter 'w'" ?

Reply:
One thing, the `get_step()` is not that nice here, maybe we could do that in the same way as in put? I.e. add inferring the step to the observe method? (We can keep the overloaded version that accepts a step, but I feel it would be verbose to do that in here)

Reply:
@karlnapf I've updated these calls. Do you still have doubts about them? :)